---
title: "Project workflow"
output: html_document
---

# 1. Download the hg19 m6a base table from RMbase website

wget http://mirlab.sysu.edu.cn/rmbase/download/human/RMBase_hg19_all_m6A_site.txt

# 2. Modified the original table - remove the top lines and # symbol

# 3. Make a script to output the GRange tables for storing the necessary information
```{r}
library(GenomicRanges)
makegranges = function(df){
  chr <- df$chromosome
  modstart = df$modStart
  modId = df$modId
  strand = df$strand
  supnum = df$supportNum
  gr  = GRanges(seqnames = chr, IRanges(start = modstart, width = 2), strand = strand)
  gr
}
```

# 4. Import the m6a sites information into each table
```{r}
library(readr)
allsites<-read_tsv("/home/xiangyu/Desktop/Co-methylation Network Project/RMBase_hg19_all_m6A_site.txt")
as.data.frame(allsites)
table(allsites$chromosome)
chr1sitespos <- allsites[which(allsites$chromosome == "chr1" & allsites$strand == "+"),]
chr1sitespos <- makegranges(df = chr1sitespos)
chr1sitesnag <- allsites[which(allsites$chromosome == "chr1" & allsites$strand == "-"),]
chr1sitesnag <- makegranges(df = chr1sitesnag)
chr2sitespos <- allsites[which(allsites$chromosome == "chr2" & allsites$strand == "+"),]
chr2sitespos <- makegranges(df = chr2sitespos)
chr2sitesnag <- allsites[which(allsites$chromosome == "chr2" & allsites$strand == "-"),]
chr2sitesnag <- makegranges(df = chr2sitesnag)
chr3sitespos <- allsites[which(allsites$chromosome == "chr3" & allsites$strand == "+"),]
chr3sitespos <- makegranges(df = chr3sitespos)
chr3sitesnag <- allsites[which(allsites$chromosome == "chr3" & allsites$strand == "-"),]
chr3sitesnag <- makegranges(df = chr3sitesnag)
chr4sitespos <- allsites[which(allsites$chromosome == "chr4" & allsites$strand == "+"),]
chr4sitespos <- makegranges(df = chr4sitespos)
chr4sitesnag <- allsites[which(allsites$chromosome == "chr4" & allsites$strand == "-"),]
chr4sitesnag <- makegranges(df = chr4sitesnag)
chr5sitespos <- allsites[which(allsites$chromosome == "chr5" & allsites$strand == "+"),]
chr5sitespos <- makegranges(df = chr5sitespos)
chr5sitesnag <- allsites[which(allsites$chromosome == "chr5" & allsites$strand == "-"),]
chr5sitesnag <- makegranges(df = chr5sitesnag)
chr6sitespos <- allsites[which(allsites$chromosome == "chr6" & allsites$strand == "+"),]
chr6sitespos <- makegranges(df = chr6sitespos)
chr6sitesnag <- allsites[which(allsites$chromosome == "chr6" & allsites$strand == "-"),]
chr6sitesnag <- makegranges(df = chr6sitesnag)
chr7sitespos <- allsites[which(allsites$chromosome == "chr7" & allsites$strand == "+"),]
chr7sitespos <- makegranges(df = chr7sitespos)
chr7sitesnag <- allsites[which(allsites$chromosome == "chr7" & allsites$strand == "-"),]
chr7sitesnag <- makegranges(df = chr7sitesnag)
chr8sitespos <- allsites[which(allsites$chromosome == "chr8" & allsites$strand == "+"),]
chr8sitespos <- makegranges(df = chr8sitespos)
chr8sitesnag <- allsites[which(allsites$chromosome == "chr8" & allsites$strand == "-"),]
chr8sitesnag <- makegranges(df = chr8sitesnag)
chr9sitespos <- allsites[which(allsites$chromosome == "chr9" & allsites$strand == "+"),]
chr9sitespos <- makegranges(df = chr9sitespos)
chr9sitesnag <- allsites[which(allsites$chromosome == "chr9" & allsites$strand == "-"),]
chr9sitesnag <- makegranges(df = chr9sitesnag)
chr10sitespos <- allsites[which(allsites$chromosome == "chr10" & allsites$strand == "+"),]
chr10sitespos <- makegranges(df = chr10sitespos)
chr10sitesnag <- allsites[which(allsites$chromosome == "chr10" & allsites$strand == "-"),]
chr10sitesnag <- makegranges(df = chr10sitesnag)
chr11sitespos <- allsites[which(allsites$chromosome == "chr11" & allsites$strand == "+"),]
chr11sitespos <- makegranges(df = chr11sitespos)
chr11sitesnag <- allsites[which(allsites$chromosome == "chr11" & allsites$strand == "-"),]
chr11sitesnag <- makegranges(df = chr11sitesnag)
chr12sitespos <- allsites[which(allsites$chromosome == "chr12" & allsites$strand == "+"),]
chr12sitespos <- makegranges(df = chr12sitespos)
chr12sitesnag <- allsites[which(allsites$chromosome == "chr12" & allsites$strand == "-"),]
chr12sitesnag <- makegranges(df = chr12sitesnag)
chr13sitespos <- allsites[which(allsites$chromosome == "chr13" & allsites$strand == "+"),]
chr13sitespos <- makegranges(df = chr13sitespos)
chr13sitesnag <- allsites[which(allsites$chromosome == "chr13" & allsites$strand == "-"),]
chr13sitesnag <- makegranges(df = chr13sitesnag)
chr14sitespos <- allsites[which(allsites$chromosome == "chr14" & allsites$strand == "+"),]
chr14sitespos <- makegranges(df = chr14sitespos)
chr14sitesnag <- allsites[which(allsites$chromosome == "chr14" & allsites$strand == "-"),]
chr14sitesnag <- makegranges(df = chr14sitesnag)
chr15sitespos <- allsites[which(allsites$chromosome == "chr15" & allsites$strand == "+"),]
chr15sitespos <- makegranges(df = chr15sitespos)
chr15sitesnag <- allsites[which(allsites$chromosome == "chr15" & allsites$strand == "-"),]
chr15sitesnag <- makegranges(df = chr15sitesnag)
chr16sitespos <- allsites[which(allsites$chromosome == "chr16" & allsites$strand == "+"),]
chr16sitespos <- makegranges(df = chr16sitespos)
chr16sitesnag <- allsites[which(allsites$chromosome == "chr16" & allsites$strand == "-"),]
chr16sitesnag <- makegranges(df = chr16sitesnag)
chr17sitespos <- allsites[which(allsites$chromosome == "chr17" & allsites$strand == "+"),]
chr17sitespos <- makegranges(df = chr17sitespos)
chr17sitesnag <- allsites[which(allsites$chromosome == "chr17" & allsites$strand == "-"),]
chr17sitesnag <- makegranges(df = chr17sitesnag)
chr18sitespos <- allsites[which(allsites$chromosome == "chr18" & allsites$strand == "+"),]
chr18sitespos <- makegranges(df = chr18sitespos)
chr18sitesnag <- allsites[which(allsites$chromosome == "chr18" & allsites$strand == "-"),]
chr18sitesnag <- makegranges(df = chr18sitesnag)
chr19sitespos <- allsites[which(allsites$chromosome == "chr19" & allsites$strand == "+"),]
chr19sitespos <- makegranges(df = chr19sitespos)
chr19sitesnag <- allsites[which(allsites$chromosome == "chr19" & allsites$strand == "-"),]
chr19sitesnag <- makegranges(df = chr19sitesnag)
chr20sitespos <- allsites[which(allsites$chromosome == "chr20" & allsites$strand == "+"),]
chr20sitespos <- makegranges(df = chr20sitespos)
chr20sitesnag <- allsites[which(allsites$chromosome == "chr20" & allsites$strand == "-"),]
chr20sitesnag <- makegranges(df = chr20sitesnag)
chr21sitespos <- allsites[which(allsites$chromosome == "chr21" & allsites$strand == "+"),]
chr21sitespos <- makegranges(df = chr21sitespos)
chr21sitesnag <- allsites[which(allsites$chromosome == "chr21" & allsites$strand == "-"),]
chr21sitesnag <- makegranges(df = chr21sitesnag)
chr22sitespos <- allsites[which(allsites$chromosome == "chr22" & allsites$strand == "+"),]
chr22sitespos <- makegranges(df = chr22sitespos)
chr22sitesnag <- allsites[which(allsites$chromosome == "chr22" & allsites$strand == "-"),]
chr22sitesnag <- makegranges(df = chr22sitesnag)
chrMsitespos <- allsites[which(allsites$chromosome == "chrM" & allsites$strand == "+"),]
chrMsitespos <- makegranges(df = chrMsitespos)
chrMsitesnag <- allsites[which(allsites$chromosome == "chrM" & allsites$strand == "-"),]
chrMsitesnag <- makegranges(df = chrMsitesnag)
chrXsitespos <- allsites[which(allsites$chromosome == "chrX" & allsites$strand == "+"),]
chrXsitespos <- makegranges(df = chrXsitespos)
chrXsitesnag <- allsites[which(allsites$chromosome == "chrX" & allsites$strand == "-"),]
chrXsitesnag <- makegranges(df = chrXsitesnag)
chrYsitespos <- allsites[which(allsites$chromosome == "chrY" & allsites$strand == "+"),]
chrYsitespos <- makegranges(df = chrYsitespos)
chrYsitesnag <- allsites[which(allsites$chromosome == "chrY" & allsites$strand == "-"),]
chrYsitesnag <- makegranges(df = chrYsitesnag)
```

# 5. Create the bins of all the methylation sites
```{r}
binchr1pos <- resize(chr1sitespos, 200, fix = "center")
binchr1nag <- resize(chr1sitesnag, 200, fix = "center")
binchr2pos <- resize(chr2sitespos, 200, fix = "center")
binchr2nag <- resize(chr2sitesnag, 200, fix = "center")
binchr3pos <- resize(chr3sitespos, 200, fix = "center")
binchr3nag <- resize(chr3sitesnag, 200, fix = "center")
binchr4pos <- resize(chr4sitespos, 200, fix = "center")
binchr4nag <- resize(chr4sitesnag, 200, fix = "center")
binchr5pos <- resize(chr5sitespos, 200, fix = "center")
binchr5nag <- resize(chr5sitesnag, 200, fix = "center")
binchr6pos <- resize(chr6sitespos, 200, fix = "center")
binchr6nag <- resize(chr6sitesnag, 200, fix = "center")
binchr7pos <- resize(chr7sitespos, 200, fix = "center")
binchr7nag <- resize(chr7sitesnag, 200, fix = "center")
binchr8pos <- resize(chr8sitespos, 200, fix = "center")
binchr8nag <- resize(chr8sitesnag, 200, fix = "center")
binchr9pos <- resize(chr9sitespos, 200, fix = "center")
binchr9nag <- resize(chr9sitesnag, 200, fix = "center")
binchr10pos <- resize(chr10sitespos, 200, fix = "center")
binchr10nag <- resize(chr10sitesnag, 200, fix = "center")
binchr11pos <- resize(chr11sitespos, 200, fix = "center")
binchr11nag <- resize(chr11sitesnag, 200, fix = "center")
binchr12pos <- resize(chr12sitespos, 200, fix = "center")
binchr12nag <- resize(chr12sitesnag, 200, fix = "center")
binchr13pos <- resize(chr13sitespos, 200, fix = "center")
binchr13nag <- resize(chr13sitesnag, 200, fix = "center")
binchr14pos <- resize(chr14sitespos, 200, fix = "center")
binchr14nag <- resize(chr14sitesnag, 200, fix = "center")
binchr15pos <- resize(chr15sitespos, 200, fix = "center")
binchr15nag <- resize(chr15sitesnag, 200, fix = "center")
binchr16pos <- resize(chr16sitespos, 200, fix = "center")
binchr16nag <- resize(chr16sitesnag, 200, fix = "center")
binchr17pos <- resize(chr17sitespos, 200, fix = "center")
binchr17nag <- resize(chr17sitesnag, 200, fix = "center")
binchr18pos <- resize(chr18sitespos, 200, fix = "center")
binchr18nag <- resize(chr18sitesnag, 200, fix = "center")
binchr19pos <- resize(chr19sitespos, 200, fix = "center")
binchr19nag <- resize(chr19sitesnag, 200, fix = "center")
binchr20pos <- resize(chr20sitespos, 200, fix = "center")
binchr20nag <- resize(chr20sitesnag, 200, fix = "center")
binchr21pos <- resize(chr21sitespos, 200, fix = "center")
binchr21nag <- resize(chr21sitesnag, 200, fix = "center")
binchr22pos <- resize(chr22sitespos, 200, fix = "center")
binchr22nag <- resize(chr22sitesnag, 200, fix = "center")
binchrMpos <- resize(chrMsitespos, 200, fix = "center")
binchrMnag <- resize(chrMsitesnag, 200, fix = "center")
binchrXpos <- resize(chrXsitespos, 200, fix = "center")
binchrXnag <- resize(chrXsitesnag, 200, fix = "center")
binchrYpos <- resize(chrYsitespos, 200, fix = "center")
binchrYnag <- resize(chrYsitesnag, 200, fix = "center")

bin <- c(binchr1pos, binchr1nag, binchr2pos, binchr2nag, binchr3pos, binchr3nag, binchr4pos, 
         binchr4nag, binchr5pos, binchr5nag, binchr6pos, binchr6nag, binchr7pos, binchr7nag, 
         binchr8pos, binchr8nag, binchr9pos, binchr9nag, binchr10pos, binchr10nag, binchr11pos, 
         binchr11nag, binchr12pos, binchr12nag, binchr13pos, binchr13nag, binchr14pos, binchr14nag, 
         binchr15pos, binchr15nag, binchr16pos, binchr16nag, binchr17pos, binchr17nag, binchr18pos, 
         binchr18nag, binchr19pos, binchr19nag, binchr20pos, binchr20nag, binchr21pos, binchr21nag, 
         binchr22pos, binchr22nag, binchrMpos, binchrMnag, binchrXpos, binchrXnag, binchrYpos, binchrYnag)

write_rds(bin, "bin.rds")
```

# 6. upload the "bin.rds" on the server, and calculate RPKM on it.
### library(readr)
### bin <- read_rds("bin.rds")
### bin
### library(GenomicAlignments)
### aln <- readGAlignments("p001_HEK293T_S1_SYSY_ip_accept_hits.bam")
### rpkm <- countOverlaps(bin, aln)/((sum(width(reduce(bin)))/10^3)*(length(aln)/10^6))

